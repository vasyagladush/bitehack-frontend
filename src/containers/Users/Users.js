import React,{useEffect, useState} from 'react';
import axios from 'axios';
import User from '../../components/User/User';
import Emoji from '../../components/Emoji/Emoji';
import * as actions from '../../store/actions/index';
import {connect} from 'react-redux';
import {url} from '../../ApiUrl';

const Users = props =>{
    const [users,setUsers] = useState([]);
    useEffect(()=>{ //component did mount setting users with our friends
        axios.get(url+'/user/friends',{headers: {
            'Authorization': 'Bearer '+localStorage.getItem('token')
        }})
        .then(response =>{
            setUsers(response.data.friends);
        })
        .catch(err=>{
            console.log(err);
        });
    },[]);
    const setUser = personId =>{ // setting user conversation after click on it in friends tab
        props.setUser(personId);
        axios.get(url+'/user/conversation',{headers:{
            "person2": personId,
            "Authorization": "Bearer "+ localStorage.getItem('token')
        }})
        .then(response=>{
           props.setMessages(response.data.messages);
        })
        .catch(err=>{
            console.log(err);
        })
    };
    let users_not_accepted = null;
    return (
        <>
        {users.length ? users.map(friend=> {
            if (friend.accepted===true) // if we have friends which has accepted status
            {
            return (
                <User key={friend.friend._id} click={()=>{setUser(friend.friend._id.toString())}} id={friend.friend._id} > {
                    // to fire onClick event on <p> component lol
                }
                    {friend.friend.name} {friend.friend.surname}
                </User>
            );
            }
            else
            {
                users_not_accepted++;
                if(users.length===users_not_accepted) // avoiding multiple users who is not accepted
                {
                    return(<p style={{fontWeight: 'bold',textAlign: 'center',fontSize: 25, marginTop: '50%'}}>You don't have friends<Emoji symbol="ðŸ˜®"/> Add them!<Emoji symbol="ðŸ¤¯"/></p>);
                }
            }
            return null;
        }) : <p style={{fontWeight: 'bold',textAlign: 'center',fontSize: 25, marginTop: '50%'}}>You don't have friends<Emoji symbol="ðŸ˜®"/> Add them!<Emoji symbol="ðŸ¤¯"/></p>}
        </>
    )
}

const mapDispatchToProps = dispatch =>{
    return {
        setUser: (personId)=>dispatch(actions.setUser(personId)),
        setMessages: (messages)=>dispatch(actions.setMessages(messages))
    }
};

export default connect(null,mapDispatchToProps)(Users);