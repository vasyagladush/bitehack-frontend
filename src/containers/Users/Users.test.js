import Users from './Users';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import {configure,mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';


const mockStore = configureMockStore();
const store = mockStore({});
//const Token= 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVmM2MzYzYwZjkyZjNhMWUzYzc1ZjIwZiIsImlhdCI6MTU5ODM0ODEyMn0.lZFeZwwL0kevnI--7bKmY8tEX1W5Vwx2zedwvqneCws'; infinite token
configure({adapter: new Adapter});

describe('<Users/>',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = mount( // if we have Provider we must use mount lol
        <Provider store={store}>
            <Users/>
        </Provider>
        );
    });
    it('matches snapshot',()=>{
        wrapper = renderer.create(<Provider store={store}>
            <Users/>
        </Provider>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
    it('should render a paragraph when we have no friends',()=>{
        expect(wrapper.find('p')).toHaveLength(1);
    });
    it('should render a user if we have friends',()=>{
        const getUser = jest.fn();
        const handleRender = jest.spyOn(React,"useEffect");
        handleRender.mockImplementation(user=>[user,getUser]);
        wrapper.update();
        expect(getUser).toBeTruthy(); // should pass our user if we have token and etc...
    });
    /*it("should update state on click", () => {
   const changeSize = jest.fn();
   const wrapper = mount(<App onClick={changeSize} />);
   const handleClick = jest.spyOn(React, "useState");
   handleClick.mockImplementation(size => [size, changeSize]);

   wrapper.find("#para1").simulate("click");
   expect(changeSize).toBeTruthy();
   */ // should be helpful
 });