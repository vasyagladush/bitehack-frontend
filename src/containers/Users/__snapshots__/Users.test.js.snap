// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users/> matches snapshot 1`] = `
<p
  style={
    Object {
      "fontSize": 25,
      "fontWeight": "bold",
      "marginTop": "50%",
      "textAlign": "center",
    }
  }
>
  You don't have friends
  <span
    aria-hidden="true"
    aria-label=""
    className="emoji"
    role="img"
  >
    ðŸ˜®
  </span>
   Add them!
  <span
    aria-hidden="true"
    aria-label=""
    className="emoji"
    role="img"
  >
    ðŸ¤¯
  </span>
</p>
`;
